integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -259,6 +236,26 @@ def test_hash_multiple_mode_query_avg_distincts(data_gen, conf):
         conf=conf)
 
 
+@approximate_float
+@ignore_order
+@incompat
+@pytest.mark.parametrize('data_gen', _init_list_no_nans, ids=idfn)
+@pytest.mark.parametrize('conf', get_params(_confs, params_markers_for_confs),
+                         ids=idfn)
+def test_hash_query_avg_multiple_distincts_with_non_distinct(data_gen, conf):
+    assert_gpu_and_cpu_are_equal_sql(
+        lambda spark : gen_df(spark, data_gen, length=100),
+        "hash_agg_table",
+        'select avg(a),' +
+        'avg(distinct b),' +
+        'avg(distinct c),' +
+        'count(a),' +
+        'sum(a),' +
+        'min(a),'+
+        'max(a) from hash_agg_table group by a',
+        conf)
+
```

Comment:
Do we have enough tests to trigger both execs for this bug with count and avg distinct?

Reply:
Yes, I verified it locally.  It goes to `GpuSortAggregateExec` path when the input is `string`. For all other input types, it takes the GpuHashAggregateExec.  In one test we have 2 countDistinct and in another test there are 2 avg distinct. I think we have covered all the cases.

Reply:
cool!